test_conformance/images/clFillImage/test_fill_generic.cpp

```diff
@@ -70,6 +74,26 @@ cl_mem create_image( cl_context context, cl_command_queue queue, BufferOwningPtr
             if ( gEnablePitch )
                 host_ptr = malloc( imageInfo->arraySize * imageInfo->slicePitch );
             break;
+        case CL_MEM_OBJECT_IMAGE1D_BUFFER:
+            if (gDebugTrace)
+                log_info(" - Creating 1D buffer image %d ...\n",
+                         (int)imageInfo->width);
+            if (gEnablePitch) host_ptr = malloc(imageInfo->rowPitch);
+            {
+                cl_int err;
+                cl_mem buffer =
+                    clCreateBuffer(context, CL_MEM_READ_WRITE,
+                                   imageInfo->rowPitch, host_ptr, &err);
+                if (err != CL_SUCCESS)
+                {
+                    log_error("ERROR: Could not create buffer for 1D buffer "
+                              "image. %ld bytes\n",
+                              imageInfo->rowPitch);
+                    return NULL;
+                }
+                imageDesc.buffer = buffer;
+            }
+            break;
```

Comment:
This code doesn't seem to be handling the usePitches case for 1D image buffer correctly. The spec requires that the buffer be created with use_host_ptr and a host_ptr but the image object inherit those from the buffer, meaning the host_ptr in that case should be NULL. 
<The CL_MEM_USE_HOST_PTR, CL_MEM_ALLOC_HOST_PTR and CL_MEM_COPY_HOST_PTR values cannot be specified in flags but are inherited from the corresponding memory access qualifiers associated with mem_object>